h2o-algos/build.gradle

```diff
@@ -13,8 +13,10 @@ dependencies {
   // Jama dependencies
   compile "gov.nist.math:jama:1.0.3"
   // netlib-java / MTJ dependencies
-  compile 'com.github.fommil.netlib:all:1.1.2'
-  compile('com.googlecode.matrix-toolkits-java:mtj:1.0.4') { exclude group: 'com.github.fommil.netlib' }
+  compile('com.github.fommil.netlib:core:1.1.2')
```

Comment:
@terrytangyuan If I'm not wrong, we just need `core` module of this lib. Previously, we were fetching also native libraries which was unnecessary.

We should always specify the exact dependencies we want to minimize libraries collisions and final assembly jar size

Reply:
Oh I see. Sure, as long as CI is happy with it I think it's fine. Thanks!

Reply:
@jakubhava we do need the native libraries otherwise we cannot use the right backend, it will compile and run but it will most likely not behave as intended

Can you please be specific about what is excluded?

Reply:
@michalkurka I probably don't understand. If tests pass, we're good to go right?

You can see what is explicitly excluded, included already:

```
compile('com.github.fommil.netlib:core:1.1.2')
compile('com.googlecode.matrix-toolkits-java:mtj:1.0.4'){
      exclude group: 'com.github.fommil.netlib', module: 'all'
 }
```

Reply:
You mentioned that you are excluding the native libraries. I was wondering which ones are excluded.

My suspicion is that we are getting rid of the libraries that would enable GPU or CPU optimizations and we will be using a Java fallback.

I am not sure we have tests for all the cases.

Forgive me for not researching this more and just blindly asking. If I get rid of the GPU backend in XGBoost the tests will still pass they will just run slower. We don't have perfect tests even there. So I am worried the tests are passing but the behavior is changed.

I promise I will look into this more when I am done with my current task.

Reply:
Ah, I see, makes sense. I probably did wrong move here and removing of the libraries is not what I wanted to do initially. This change should be just syntax sugar, but should not change any semantics.

I will try to fix this with having ^ in mind.

Thanks for the reviews!

Reply:
```
compile('com.github.fommil.netlib:all:1.1.2')
compile('com.googlecode.matrix-toolkits-java:mtj:1.0.4')
``` In this case it's semantically same as original code as we are having the same libraries. The exclusion is not necessary as the matrix-toolkits depends on the same version of the library.

This will probably works as expected. DBC resolver is confused in this case:
```
compile('com.github.fommil.netlib:all:1.1.2')
compile('com.googlecode.matrix-toolkits-java:mtj:1.0.4'){
exclude  group: 'com.github.fommil.netlib', module: '1.1.2'
}
``` It thinks we want to exclude `com.github.fommil.netlib` globally what we don't want do to

Reply:
@michalkurka good news!! the `all` artifact was problematic. I was ably to achieve solution which uses completely same dependencies (manual, but working solution - explicitly list all dependencies in `all` artifact), but DBC is happy as well. So no changes regarding libraries here 👍